### PR TITLE
Omit the value attribute from select options with no value

### LINKED ISSUE #3440: Possibly unexpected behaviour if you provide text but no value for select items
## Description of the issue

possible foot gun when using govukSelect component and not passing an explicit value to your select items

https://ukgovernmentdigital.slack.com/archives/C6DMEH5R6/p1680092442731649

https://github.com/alphagov/govuk-design-system-backlog/issues/60

## Steps to reproduce the issue

pass only the text parameter to select items when using govukSelect

## Actual vs expected behaviour

If you don't supply a value to a select option `<option>text</option>` then the value of that form field will be "text"

However, with govukSelect, [if you don't supply a value for one of the select items then you'll get an empty value attribute](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/select/template.njk#L47) `<option value="">text</option>` which will cause the value of that form field to (maybe unexpectedly) be an empty string ""

I can see the value of being able to pass an empty string as a value, but it seems like it should be an explicit choice rather than the default

## Environment (where applicable)

N/A

- Operating system:
- Browser:
- Browser version:
- GOV.UK Frontend Version:

### PR DESCRIPTION
**Pull request by 36degrees**
In HTML, if you omit the `value` attribute from an `<option>` [the value defaults to the text contained inside the element][1].

However if you omit the value from an option when using the `govukSelect` macro, we currently include an empty `value` attribute on the `<option>` element. This would mean that when submitting a form using a select with options with no value set, the value submitted would be the empty string.

Instead, omit the `value` attribute if no value option has been provided. We then mimic the default HTML behaviour, as the option value will then naturally fall back to the text content.

We still need to ensure that the `value` attribute is included when the value option is passed but has a falsey value (like an empty string, 0, or boolean false).

Also allow the top-level `value` option on a `govukSelect` to work with options that have no explicit value set by comparing it to the text content instead.

Fixes #3440.

[1]: https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element:~:text=The%20value%20attribute%20provides%20a%20value%20for%20element.%20The%20value%20of%20an%20option%20element%20is%20the%20value%20of%20the%20value%20content%20attribute%2C%20if%20there%20is%20one%2C%20or%2C%20if%20there%20is%20not%2C%20the%20value%20of%20the%20element%27s%20text%20IDL%20attribute.

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/select/template.test.js [hunk 1]
```diff
@@ -33,13 +33,43 @@ describe('Select', () => {
...
+      const $thirdItem = $('.govuk-select option:nth(2)')
+      expect($thirdItem.attr('value')).toEqual('0')
+    })
+
+    it('omits the value attribute if no value option is provided', () => {
+      const $ = render('select', examples['without values'])
+
+      const $firstItem = $('.govuk-select option:first-child')
+      // Ideally we'd test for $firstItem.attr('value') == undefined but it's
+      // broken in Cheerio – https://github.com/cheeriojs/cheerio/issues/3237
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/select/template.test.js:69]: ❤️

File: packages/govuk-frontend/src/govuk/components/select/template.test.js [hunk 2]
```diff
@@ -61,6 +91,13 @@ describe('Select', () => {
       expect($selectedItem.attr('selected')).toBeTruthy()
     })
 
+    it('selects options with implicit value using selected value', () => {
+      const $ = render('select', examples['without values with selected value'])
+
+      const $selectedItem = $('option:contains(\'Green\')')
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/select/template.test.js:97]: ESLint should be happy with double quotes when they're intentionally nesting single quotes if you like

```suggestion
      const $selectedItem = $("option:contains('Green')")
```

**Review by colinrotherham — APPROVED** [review]

**Comment by colinrotherham** [comment]
@36degrees Do we need a CHANGELOG entry for this should anyone rely on the old behaviour?

**Comment by 36degrees** [comment]
> Do we need a CHANGELOG entry for this should anyone rely on the old behaviour?

D'oh! I'd added it but not pushed. Have applied your suggestion and updated.

**Review by colinrotherham — APPROVED** [review]
Looking good still 😊

**Pull request merged**